rest/data/project_settings_test.go

```diff
@@ -0,0 +1,391 @@
+			pRefFromDB, err := model.FindBranchProjectRef(ref.Id)
+			assert.NoError(t, err)
+			assert.NotNil(t, pRefFromDB)
+			assert.NotEmpty(t, pRefFromDB.SpawnHostScriptPath)
+		},
+		model.ProjectPageAccessSection: func(t *testing.T, ref model.ProjectRef) {
+			newAdmin := user.DBUser{
+				Id: "newAdmin",
+			}
+			require.NoError(t, newAdmin.Insert())
+			ref.Restricted = nil // should now default to the repo value
+			ref.Admins = []string{"oldAdmin", newAdmin.Id}
+			apiProjectRef := restModel.APIProjectRef{}
+			assert.NoError(t, apiProjectRef.BuildFromService(ref))
+			apiChanges := &restModel.APIProjectSettings{
+				ProjectRef: apiProjectRef,
+			}
+			assert.NoError(t, dc.SaveProjectSettingsForSection(ctx, ref.Id, apiChanges, model.ProjectPageAccessSection, false, "me"))
+			pRefFromDB, err := model.FindBranchProjectRef(ref.Id)
+			assert.NoError(t, err)
+			assert.NotNil(t, pRefFromDB)
+			assert.Nil(t, pRefFromDB.Restricted)
+			assert.Equal(t, pRefFromDB.Admins, ref.Admins)
+
+			mergedProject, err := model.FindMergedProjectRef(ref.Id)
+			assert.NoError(t, err)
+			assert.NotNil(t, mergedProject)
+			assert.True(t, mergedProject.IsRestricted())
+
+			restrictedScope, err := rm.GetScope(ctx, evergreen.RestrictedProjectsScope)
+			assert.NoError(t, err)
+			assert.NotNil(t, restrictedScope)
+			assert.Contains(t, restrictedScope.Resources, ref.Id)
+
+			unrestrictedScope, err := rm.GetScope(ctx, evergreen.UnrestrictedProjectsScope)
+			assert.NoError(t, err)
+			assert.NotNil(t, unrestrictedScope)
+			assert.NotContains(t, unrestrictedScope.Resources, ref.Id)
+
+			newAdminFromDB, err := user.FindOneById("newAdmin")
+			assert.NoError(t, err)
+			assert.NotNil(t, newAdminFromDB)
+			assert.Contains(t, newAdminFromDB.Roles(), "admin")
+		},
+		model.ProjectPageVariablesSection: func(t *testing.T, ref model.ProjectRef) {
+			// remove a variable, modify a variable, add a variable
+			updatedVars := &model.ProjectVars{
+				Id:          ref.Id,
+				Vars:        map[string]string{"it": "me", "banana": "phone"},
+				PrivateVars: map[string]bool{"banana": true},
+			}
+			apiProjectVars := restModel.APIProjectVars{}
+			assert.NoError(t, apiProjectVars.BuildFromService(updatedVars))
+			apiChanges := &restModel.APIProjectSettings{
+				Vars: apiProjectVars,
+			}
+			assert.NoError(t, dc.SaveProjectSettingsForSection(ctx, ref.Id, apiChanges, model.ProjectPageVariablesSection, false, "me"))
+			varsFromDb, err := model.FindOneProjectVars(updatedVars.Id)
+			assert.NoError(t, err)
+			assert.NotNil(t, varsFromDb)
+			assert.Equal(t, varsFromDb.Vars["it"], "me")
+			assert.Equal(t, varsFromDb.Vars["banana"], "phone")
+			assert.Equal(t, varsFromDb.Vars["hello"], "")
+			assert.False(t, varsFromDb.PrivateVars["it"])
+			assert.False(t, varsFromDb.PrivateVars["hello"])
+			assert.True(t, varsFromDb.PrivateVars["banana"])
+		},
+		model.ProjectPageNotificationsSection: func(t *testing.T, ref model.ProjectRef) {
```

Comment:
This is the only thing I added to this file (and the subscription added  to the DB below)